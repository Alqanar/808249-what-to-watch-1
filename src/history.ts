import {createBrowserHistory} from "history";


declare const process: {
  env: {
    NODE_ENV: string;
  };
};

const baseName = process.env.NODE_ENV === `production` ? `/what-to-watch` : `/`;

export default createBrowserHistory({basename: baseName});
