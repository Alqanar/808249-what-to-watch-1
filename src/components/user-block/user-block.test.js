import React from "react";
import renderer from "react-test-renderer";
import {Router} from "react-router-dom";

import UserBlock from "./user-block.tsx";
import history from "../../history.ts";


describe(`User block correctly renders`, () => {
  it(`if isAuth = true renders avatar block`, () => {
    const tree = renderer
      .create(
        <Router history={history}>
          <UserBlock
            avatarLink="img/avatar.jpg"
            isAuth={true}
          />
        </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`if isAuth = false renders sign in link`, () => {
    const tree = renderer
      .create(
        <Router history={history}>
          <UserBlock
            avatarLink=""
            isAuth={false}
          />
        </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
