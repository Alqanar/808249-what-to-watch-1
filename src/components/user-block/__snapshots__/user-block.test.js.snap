// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User block correctly renders if isAuth = false renders sign in link 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="#"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;

exports[`User block correctly renders if isAuth = true renders avatar block 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <img
      alt="User avatar"
      height="63"
      src="img/avatar.jpg"
      width="63"
    />
  </div>
</div>
`;
